### PR TITLE
Upgrade emotion

### PR DESCRIPTION
**Pull request by Chrischuck**
v10 -> v11

Now works out of box w/ Next.js; no need to add anything in `_document.js`

To test, make sure you `yarn link` `bumbag-server` and `gatsby-plugin-bumbag`

### CODE REVIEW

File: packages/bumbag-addon-highlighted-code/jest.config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-const { ...baseConfig } = require('../../jest.config');
+const { ...baseConfig } = require("../../jest.config");
 
 module.exports = {
   ...baseConfig,
-  snapshotSerializers: ['jest-emotion']
+  snapshotSerializers: ["@emotion/jest"],
```
Comment (jxom) [inline packages/bumbag-addon-highlighted-code/jest.config.js:5]: Looks like all the changed files are being replaced with double quotes. Can we run `yarn lint` to fix this?

**Review by jxom — CHANGES_REQUESTED** [review]
Thanks for the PR! I remember we migrated to Emotion 11 when it first came out a few months ago, however, we ran into some issues with specificity when using CSS Props along with a `styled` component. But when testing this PR, it seems to be okay now (maybe emotion fixed something).

Can you also run the tests as well? I would expect some snapshots to update along with this migration.

File: packages/bumbag-addon-highlighted-code/jest.config.js (hunk 1 above)
Comment (Chrischuck) [reply packages/bumbag-addon-highlighted-code/jest.config.js:5]: Will do, seems like my global prettier config changing that, thoughts on adding a .prettierrc/prettier to the codebase?

Comment (jxom) [reply packages/bumbag-addon-highlighted-code/jest.config.js:5]: We already have our prettier config in our ESLint config (https://github.com/bumbag/bumbag-ui/blob/master/.eslintrc#L9). So if you're using the ESLint extension for VSCode, you should be able to do `cmd-shift-p` -> `ESLint: Fix all auto-fixable problems`

File: packages/bumbag/src/Modal/Modal.tsx [hunk 2]
```diff
@@ -1,14 +1,26 @@
-import * as React from 'react';
-import { Box as ReakitBox, DialogProps as ReakitDialogProps, useDialog as useReakitDialog } from 'reakit';
-import _merge from 'lodash/merge';
+import * as React from "react";
```
Comment (jxom) [inline packages/bumbag/src/Modal/Modal.tsx:1]: Can we format this file? :)

File: packages/bumbag/src/Provider/Provider.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { Provider as ReakitProvider } from 'reakit';
+import { ThemeProvider as EmotionProvider } from '@emotion/react';
```
Comment (jxom) [inline packages/bumbag/src/Provider/Provider.tsx:3]: Can we not import ThemeProvider from `../styled`?

File: packages/bumbag/src/styled/index.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export { default as classNames } from 'classnames';
-export { flush, hydrate, cx, getRegisteredStyles, injectGlobal, sheet, cache, css as cssClass } from 'emotion';
-export { css, keyframes, Global, ThemeContext, CacheProvider } from '@emotion/core';
-export { withTheme, ThemeProvider } from 'emotion-theming';
+export { flush, hydrate, cx, getRegisteredStyles, injectGlobal, sheet, css as cssClass, cache } from '@emotion/css';
+export { css, keyframes, Global, ThemeContext, CacheProvider } from '@emotion/react';
```
Comment (jxom) [inline packages/bumbag/src/styled/index.ts:3]: We can probably consolidate these lines so `withTheme` and `ThemeProvider` is exported here too.

File: packages/gatsby-plugin-bumbag/gatsby-ssr.js [hunk 5]
```diff
@@ -6,10 +6,10 @@
 
 // You can delete this file if you're not using it
 
-import React from 'react';
-import { renderToString } from 'react-dom/server';
-import { renderStylesToString } from 'emotion-server';
-import { InitializeColorMode } from 'bumbag';
+import React from "react";
```
Comment (jxom) [inline packages/gatsby-plugin-bumbag/gatsby-ssr.js:9]: Can we format this file?

**Comment by jxom** [comment]
Looking amazing @Chrischuck! Just another few comments and hopefully we can get this merged.

File: packages/gatsby-plugin-bumbag/gatsby-ssr.js (hunk 5 above)
Comment (Chrischuck) [reply packages/gatsby-plugin-bumbag/gatsby-ssr.js:9]: I think this is left out from the `yarn lint:fix` script, but fixed manually

File: packages/bumbag/src/Modal/Modal.tsx (hunk 2 above)
Comment (Chrischuck) [reply packages/bumbag/src/Modal/Modal.tsx:1]: Done. One more thing though, this component errored out when trying to update the snapshot:
Screen Shot 2021-03-14 at 21 57 02

I haven't looked too closely, but was wondering if you have a better idea of what's causing it

File: packages/bumbag/src/styled/index.ts (hunk 4 above)
Comment (Chrischuck) [reply packages/bumbag/src/styled/index.ts:3]: Thanks for the catch!

**Comment by Chrischuck** [comment]
Thanks @jxom! which comments are left? and can you take a look at the screenshot I posted of the model snapshot, wondering if you could point me in the direction of what needs fixing for it

**Comment by jxom** [comment]
Thanks! 

The failing `Modal` test seemed to be because of a `className={undefined}` prop that was added to the backdrop. I have fixed it now though.

**Pull request merged**